File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/watchproviders/model/WatchProvider.java
package dev.reuss.tmdb.domain.watchproviders.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.Map;

/**
 * TMDB watch provider.
 *
 * @param displayPriorities region-specific display priorities
 * @param displayPriority   default display priority
 * @param logoPath          provider logo image path
 * @param providerName      provider name
 * @param providerId        TMDB provider id
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record WatchProvider(
        @JsonProperty("display_priorities")
        Map<String, Integer> displayPriorities,

        @JsonProperty("display_priority")
        int displayPriority,

        @JsonProperty("logo_path")
        String logoPath,

        @JsonProperty("provider_name")
        String providerName,

        @JsonProperty("provider_id")
        int providerId
) implements TmdbModel {
    public WatchProvider {
        displayPriorities = TmdbCollections.map(displayPriorities);
    }

}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/watchproviders/model/WatchProvider.kt
package dev.reuss.tmdb.domain.watchproviders.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TMDB watch provider.
 *
 * @property displayPriorities region-specific display priorities
 * @property displayPriority default display priority
 * @property logoPath TMDB image path for the logo.
 * @property providerName provider name
 * @property providerId TMDB provider id
 */
@JvmRecord
data class WatchProvider(
    @all:JsonProperty("display_priorities")
    val displayPriorities: Map<String, Int> = emptyMap(),
    @all:JsonProperty("display_priority")
    val displayPriority: Int,
    @all:JsonProperty("logo_path")
    val logoPath: String?,
    @all:JsonProperty("provider_name")
    val providerName: String?,
    @all:JsonProperty("provider_id")
    val providerId: Int,
) : TmdbModel
